src/lib/models/emulation/Core.cc

```diff
@@ -26,117 +37,103 @@ Core::Core(memory::MemoryInterface& instructionMemory,
+          dataMemory_.requestRead(target);
+          // Save addresses for use by instructions that perform a LD and STR
+          // (i.e. single instruction atomics)
+          previousAddresses_.push_back(target);
+        }
+        // Emulation core can only be used with a Flat memory interface, so data
+        // is ready immediately
+        const auto& completedReads = dataMemory_.getCompletedReads();
+        assert(
+            completedReads.size() == addresses.size() &&
+            "Number of completed reads does not match the number of requested "
+            "reads.");
+        for (const auto& response : completedReads) {
+          uop->supplyData(response.target.address, response.data);
+        }
+        dataMemory_.clearCompletedReads();
+      }
+    } else if (uop->isStoreAddress()) {
+      auto addresses = uop->generateAddresses();
+      previousAddresses_.clear();
+      if (uop->exceptionEncountered()) {
+        handleException(uop);
+        // If fatal, return
+        if (hasHalted_) return;
+      }
+      // Store addresses for use by next store data operation in `execute()`
       for (auto const& target : addresses) {
-        dataMemory_.requestRead(target);
-        // Save addresses for use by instructions that perform a LD and STR
-        // (i.e. single instruction atomics)
         previousAddresses_.push_back(target);
       }
-      // Emulation core can only be used with a Flat memory interface, so data
-      // is ready immediately
-      assert((config::SimInfo::getConfig()["L1-Data-Memory"]["Interface-Type"]
-                  .as<std::string>() == "Flat") &&
-             "Emulation core is only compatable with a Flat Memory Interface.");
-      const auto& completedReads = dataMemory_.getCompletedReads();
-      assert(completedReads.size() == addresses.size() &&
-             "Number of completed reads does not match the number of requested "
-             "reads.");
-      for (const auto& response : completedReads) {
-        uop->supplyData(response.target.address, response.data);
+      if (!uop->isStoreData()) {
+        // No further action needed, move onto next micro-op
+        macroOp_.erase(macroOp_.begin());
+        continue;
       }
-      dataMemory_.clearCompletedReads();
-    }
-  } else if (uop->isStoreAddress()) {
-    auto addresses = uop->generateAddresses();
-    previousAddresses_.clear();
-    if (uop->exceptionEncountered()) {
-      handleException(uop);
-      return;
-    }
-    // Store addresses for use by next store data operation in `execute()`
-    for (auto const& target : addresses) {
-      previousAddresses_.push_back(target);
-    }
-    if (!uop->isStoreData()) {
-      // No further action needed, fetch memory for next cycle and return early
-      instructionMemory_.requestRead({pc_, FETCH_SIZE});
-      microOps_.pop();
-      return;
     }
+    execute(uop);
+    macroOp_.erase(macroOp_.begin());
   }
-  execute(uop);
+  // Commit
```

Comment:
Labelling this line as `Commit` seems a bit odd as nothing is functionally happening

Reply:
Removed